rtc/configuration.md

```diff
@@ -136,143 +136,198 @@ tinymce.init({
+If the RTC session fails to connect, or is disconnected due to an error, the user will be blocked from editing (using [setProgressState]({{site.baseurl}}/api/tinymce/tinymce.editor/#setprogressstate)) along with an error notification:
+
+![RTC disconnected error example]({{site.baseurl}}/images/rtc-error-example.png "RTC disconnected error example")
 
+The server disconnected callback can be used to provide an integration response to this condition. The function is provided both a reason for the disconnect and a suggested HTML error message to display to the user.
+
+> **Caution**: It is critical to at least handle the `client_update_required` reason. This indicates the plugin is out of date compared to other users on the session. The behaviour in this scenario depends on the configuration:
+> * If the `rtc_server_disconnected` is set, no message is displayed to the user for this error. It is up to the integrator to manage cleanly reloading the page.
+> * If the `rtc_server_disconnected` is not set, the suggested error message will be displayed in a notification asking the user to reload the page.
 
 **Type:** `Function`
 
 **Required:** no
 
-#### Input fields for `rtc_snapshot`
+#### Input fields for `rtc_server_disconnected`
 
 | Field | Type | Description |
 |-------|:----:|-------------|
-| `documentId` | `string` | The document ID from the [`rtc_document_id`](#rtc_document_id) option
-| `version` | `integer` | An increasing version number, specific to the current document ID, between 0 and 2147483648 (2<sup>31</sup>). |
-| `getContent()` | `string` | Function to execute to get the content for that particular version. |
+| `reason` | `string` | One of the reasons described below. |
+| `message` | `string` | A translated description suitable for displaying to users. This string might contain HTML, and in some cases is the same string displayed in the editor notification. |
 
-#### Example of getting content snapshots
+#### Reasons for disconnection
+
+The `reason` field will have one of the following values. 
+
+| Value |  Description |
+|-------|-------------|
+| `client_update_required` | This error indicates the RTC plugin is out of date and cannot connect to an active session for the supplied `rtc_document_id`. This can happen on startup, but is more common at runtime during editor upgrades. The suggested message recommends the user reload the page. |
+| `encryption` | Indicates a failure on startup either in the cryptography process or `rtc_encryption_provider`. This usually means there is an error in the editor configuration. |
+| `jwt` | Indicates a problem with `rtc_token_provider`. Either the provider returned a rejected promise or the token was invalid. |
+| `content` | Indicates a problem with `rtc_initial_content_provider`. The only time this generally happens is when the provider returns a rejected promise. |
```

Comment:
"The only" conflicts with "generally"
```suggestion
| `content` | Indicates a problem with `rtc_initial_content_provider`. This generally happens when the provider returns a rejected promise. |
```

Reply:
Weird wording, I agree. I didn't want to make it definitive in case it changes (there is actually only one reason in the code right now) but it did make it to vague. I'll change it completely.